src/app/public/src/modules/shared/search-results-provider.ts

```diff
@@ -0,0 +1,151 @@
+import { Injectable, Optional } from '@angular/core';
+import { SkyAuthHttp } from '@blackbaud/skyux-builder/runtime';
+import { HttpClient } from '@angular/common/http';
+import { StacheConfigService } from './config.service';
+import { StacheSearchResult } from './search-result';
+import 'rxjs/add/operator/toPromise';
+
+@Injectable()
+export class StacheSearchResultsProvider {
+
+  private http: any;
+  private moreResultsUrl: string;
+  private moreResultsLabel: string = 'Show More';
+  private htmlFields: any = {
+    description: true
+  };
+  private isInternal: boolean = true;
+  private siteNames: string[] = [];
+  private endpoints: any = {
+    internal: 'query',
+    public: 'query-public'
+  };
+  private queryUrlBase: string = 'https://stache-search-query.sky.blackbaud.com';
+
+  constructor(
+    @Optional() private authHttp: SkyAuthHttp,
+    private regHttp: HttpClient,
```

Comment:
Does this have to be injected like this?  We can't conditionally create a new one in the constructor itself?

Reply:
Not unless we want to do conditional injection, which is an anti-pattern.

Reply:
https://stackoverflow.com/questions/43450259/angular-2-conditionally-inject-service-in-component
? The Injector pattern?

Reply:
Yeah, I saw that post too. If you read further down, they explicitly say to avoid it. I think this solution circumvents that issue and is much more readable. I could add a comment for clarity if you think that would help. Thoughts @Blackbaud-SteveBrush?

Reply:
Just trying to see if there is an option we can use to avoid having an un-used injected class if the auth exists.
```
constructor(
@Optional() private http: SkyAuthHttp
) {
 if(!this.http) {
   this.http = <Http>this.injector.get(Http);
}
}
```

Reply:
If it's that big of a concern (and I'm not convinced it is), then really we should create a wrapper service for it and inject that instead rather than using the injector directly.

Reply:
I agree that this is messy. Ideally, you would use a [provider factory](https://stackoverflow.com/questions/40027892/angular-cli-conditional-imports-using-an-environment-variable). If this proves too difficult, I would at the very least isolate this logic into a separate service (such as `StacheSearchResultsHttp`) that can be injected into this provider's constructor. The http service would then determine which client is should use.

Also, curious why you added the suffix "Provider" to this class, instead of "Service"?

Reply:
I think the factory will be fine. We can abstract it there. Not sure how testing that will be though. Kinda seems like it will be terrible. And I'm using Provider to be in line with the SKYUX and Omnibar naming conventions for that particular thing. That's what Paul names them.